### DISCUSSION TITLE
Heltec V3 resetting on battery empty

### DISCUSSION THREAD

**Discussion by Droopydroors** [meshtastic/Discussions#20] [ESP32]
Still digging into this but the Heltec v3 board is running 2.5.5.  It is a handheld device.  When the battery dies and the device gets plugged in to charge the long and short name are reset.  It also is blocking private messages from other nodes that it previously exchanged keys with.  So i will check next time if the power outage is also resetting the public and private keys.

**Comment by ikearat** [comment]
I am the new owner of a Heltec V3 which I guess is the same thing you have. After reading all over the place and finding Nothing (yes, used a capitol letter there) about how the ESP32 board manages memory commits while in power fail, a few discussions for work-around fixes to the issue and some rabbit holes with individual builds, I decided to never let my node lose battery power. This would seem to end the set-and-forget aspect of using the Heltec ESP32 V3 as a field node.

**Reply by Droopydroors** [reply]
I only use rak nodes for the field nodes which are all solar and never have power drop below 90%.  But I love the Heltec V3 because its cheap and i can just 3d print cases and hand them out like candy to get more people involved.  But this problem kinda sucks.  It wouldnt be an issue with firmware before 2.5 mainly because if its renewing public and private keys then private messages will not go through until both sides delete the node so they can re-establish the new keys.  Not good.
